### PR TITLE
Correcting a mpi n_nodes multiplicative factor in nk.jax.expect 

### LINKED ISSUE #1355: [bug] nk.jax.expect is off by a factor of mpi.n_rank in the backward pass
it's just a multiplicative factor, but still...

discovered by Alessandro Sinibaldi

### PR DESCRIPTION
**Pull request by alleSini99**
The gradient computed with function `nk.jax.expect` misses a multiplicative factor given by the number of mpi ranks. The issue is in the function `mpi_mean` inside the definition of `f` within `_expect_bwd`, because using it we already do the mean on the different ranks inside the function itself while we should do it afterwards. This results in a rescaling of the output gradient of 1/`nk.utils.mpi.n_nodes`. Therefore, we substitute `mpi_mean` with a simple mean over the samples and we perform the mpi mean over the gradients afterwards. 

closes #1355

### CODE REVIEW

File: test/jax/test_nk_jax_expect_mpi.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        logpdf = lambda params_, sigma_: jnp.log(
+            jnp.square(jnp.absolute(jnp.exp(apply_fun({"params": params_}, sigma_))))
+        )
+
+        return nk.jax.expect(logpdf, e_loc_, params, sigma)[0]
+
+    Ē, E_vjp = nk.jax.vjp(expval_pars, params)
+    E_grad = E_vjp(jnp.ones_like(Ē))[0]
+    E_grad = jax.tree_map(lambda x: mpi_mean_jax(x)[0], E_grad)
```
Comment (PhilipVinc) [inline test/jax/test_nk_jax_expect_mpi.py:48]:
```suggestion
    E_grad = jax.tree_map(lambda x: nk.utils.mpi.mpi_mean_jax(x)[0], E_grad)
```

**Comment by alleSini99** [comment]
sono proprio bravo

**Review by PhilipVinc — APPROVED** [review]
thanks! this is finally great!
